### ISSUE TITLE
sample_weight ignored

### ISSUE THREAD

**Issue by abirkmanis** [tensorflow/tensorflow#28029]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
Yes.
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
"Ubuntu 16.04.5 LTS"
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
N/A
- TensorFlow installed from (source or binary):
Docker image from tensorflow/tensorflow:latest-py3-jupyter
- TensorFlow version (use command below):
b'v1.13.1-0-g6612da8951' 1.13.1
- Python version:
3.5.2 (default, Nov 12 2018, 13:43:14) 
[GCC 5.4.0 20160609]
- Bazel version (if compiling from source):
N/A
- GCC/Compiler version (if compiling from source):
N/A
- CUDA/cuDNN version:
N/A
- GPU model and memory:
N/A

**Describe the current behavior**
sample_weight seems to be ignored by fit(), test_on_batch(), etc.

**Describe the expected behavior**
sample_weight should affect the loss.

**Code to reproduce the issue**
```
import numpy as np
from tensorflow.keras.models import Model
from tensorflow.keras.layers import Input
from tensorflow.keras.optimizers import Adam
from tensorflow.keras.losses import mse

# a dummy model that just returns the inputs
i=Input(shape=(1,))
m=Model(inputs=[i],outputs=[i])
m.compile(optimizer=Adam(),loss=[mse])

# the unweighted loss should be 1
xs=np.zeros((1,1))
ys=np.ones((1,1))

# expecting loss to be weighted by sample_weight (so be 0.1 for the second test)
print(m.test_on_batch(xs,ys))
print(m.test_on_batch(xs,ys,np.array([.1])))
```

**Comment by abirkmanis** [comment]
In TF 2.0.0-alpha0 the behavior is as expected - that led me to believe the behavior in 1.13.1 is a bug.

**Comment by abirkmanis** [comment]
Looks like behavior changed in https://github.com/tensorflow/tensorflow/commit/bacd851cf8281b3a87d782bbaf1888df571337a6#diff-4f0d455edc07c640cbb22c8e39a61dd5.
Presuming I cannot update to a nightly, what is the workaround meanwhile?

**Comment by jvishnuvardhan** [comment]
@abirkmanis I can reproduce the issue. Output from TF2.0 is 1.0, and 0.1 whereas output from TF1.13.1 is 1.0 and 1.0. Thanks!

**Comment by omalleyt12** [comment]
Hi @abirkmanis , looks like this was fixed as you say

IIRC, previous behavior was to divide the sample_weights by the mean of the sample_weights, which is why it looks like it is being ignored in this case. My suggestion is if the mean of your sample_weights is not 1, then to either rescale the sample_weights or rescale your loss_fn in order to get the desired behavior
